File: ruby/opalcommand/tests/integration/flow_test.rb
# frozen_string_literal: true

require_relative '../test_helper'

class FlowTest < Minitest::Test
  def test_dispatch_routing_workflow_flow
    orders = OpalCommand::Core::Dispatch.plan_settlement([{ id: 's', urgency: 4, eta: '10:00' }], 1)
    route = OpalCommand::Core::Routing.choose_corridor([{ channel: 'north', latency: 4 }], [])
    assert_equal 1, orders.length
    assert_equal 'north', route[:channel]
    assert OpalCommand::Core::Workflow.transition_allowed?(:queued, :allocated)
  end
end

File: go/atlasdispatch/tests/integration/flow_test.go
package integration

import (
	"atlasdispatch/internal/allocator"
	"atlasdispatch/internal/routing"
	"atlasdispatch/internal/workflow"
	"testing"
)

func TestDispatchRoutingWorkflowFlow(t *testing.T) {
	orders := allocator.PlanDispatch([]allocator.Order{{ID: "m", Urgency: 4, ETA: "10:00"}}, 1)
	route := routing.ChooseRoute([]routing.Route{{Channel: "north", Latency: 4}}, map[string]bool{})
	if len(orders) != 1 || route == nil || route.Channel != "north" {
		t.Fatalf("unexpected flow outputs")
	}
	if !workflow.CanTransition("queued", "allocated") {
		t.Fatal("expected transition")
	}
}
